stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/ChatDomainImpl.kt

```diff
@@ -257,10 +258,9 @@ internal class ChatDomainImpl internal constructor(
             repos.configs.load()
 
             // load the current user from the db
-            val initialSyncState = SyncStateEntity(currentUser.id)
-            syncState = repos.syncState.select(currentUser.id) ?: initialSyncState
+            syncStateFlow.value = repos.syncState.select(currentUser.id) ?: SyncState(currentUser.id)
```

Comment:
This should probably go into a local variable first, as this way we're writing it, and then reading it back twice later on in this function (which might have a new value by then)

Reply:
I'm not sure how it should work (we don't have any described logic for that or tests). I did some refactoring on that part, take a look

Reply:
The refactor looks good to me, I think it makes sense this way 👍